NppExec/src/PickColorBtn.h

```diff
@@ -0,0 +1,145 @@
+/*
+This file is part of NppExec
+Copyright (C) 2023 Anders Kjersem.
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef PICKCOLORBTN_INC
+#define PICKCOLORBTN_INC
+#include <windows.h>
+#include <commdlg.h>
+#include <CommCtrl.h>
+
+
+enum PICKCOLORBTNNOTIFYCMD {
+    PCBN_INITCHOOSE = 0x8000+0,
+};
+
+#define PickColorBtn_GetColor(h) ( (COLORREF) GetWindowLongPtr((h), GWLP_USERDATA) )
+#define PickColorBtn_SetColor(h, c) ( SetWindowLongPtr((h), GWLP_USERDATA, (c)), InvalidateRect((h), NULL, false) )
+
+#ifndef PickColorBtn_FillRectColor
+static inline void PickColorBtn_FillRectColor(HDC hDC, const RECT*pR, COLORREF Clr)
+{
+    UINT orgbc = SetBkColor(hDC, Clr);
+    ExtTextOut(hDC, 0, 0, ETO_OPAQUE, pR, NULL, 0, NULL);
+    SetBkColor(hDC, orgbc);
+}
+#endif
+
+struct PickColorBtn {
+    template<class T> static INT_PTR CALLBACK HandleMessage(T hDlg, UINT Msg, WPARAM WPar, LPARAM LPar, LRESULT *RetVal)

```

Comment:
Do we really need templated methods rather than regular methods?

Reply:
I admit it is a bit ugly and seems unnecessary but since this is a header-only implementation it is actually better this way, the compiler will generate a single function that is used by all the .cpp files instead of using just `static ` and hoping that `/OPT:ICF` with the compiler will merge them all.

The alternative is to move it to a .cpp (I already had this header-only implementation laying around).

Reply:
I'd prefer non-template implementation in a separate .cpp file.